Refactoring caching utils for api using fastapi-cache  (#420)

peterdudfield opened the pull request:
# Pull Request

## Description

- use fast-api caching library

#399 

## How Has This Been Tested?

- [ ] CI tests
- [ ] ran against local database, and check cache was working
- [ ] ran 5 same command `` at the same time and average response time was ...

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

peterdudfield commented:
@yuvraajnarula ive left a TODO to do with making sure the `user` is not in the cache, but other parameters are. Would you be intereted in doing this?

yuvraajnarula commented:
yes, I'll be interested in working on it

yuvraajnarula commented:
Hi, @peterdudfield.
I removed the `user` from cache(other params are present) and refactored database.py using type hints.

peterdudfield commented:
> Hi, @peterdudfield. I removed the `user` from cache(other params are present) and refactored database.py using type hints.

Thanks, do you mind, doing separate PRs for these different issues. Just so things dont get confused

yuvraajnarula commented:
> > Hi, @peterdudfield. I removed the `user` from cache(other params are present) and refactored database.py using type hints.
> 
> Thanks, do you mind, doing separate PRs for these different issues. Just so things dont get confused

@peterdudfield I've done exactly as you said #429 for removing user from key, but making sure other params are in the cache, and #432 for database.py type hints support.
